doc/guides/porting/board_porting.rst

```diff
@@ -45,135 +31,333 @@ available in the source tree:
+started. See :ref:`custom_board_definition` for documentation on moving your
+board directory to a separate repository once it's working.)
 
-Board
-=====
+Your board directory should look like this:
 
-A board implements an SoC with all its features, together with peripherals
-available on the board that differentiates the board with additional interfaces
-and features not available in the SoC.
+.. code-block:: none
 
-A board port on Zephyr typically consists of two parts:
+   boards/<ARCH>/plank
+   ├── board.cmake
+   ├── doc
+   │   └── index.rst
+   ├── Kconfig.board
+   ├── Kconfig.defconfig
+   ├── plank_defconfig
+   ├── plank.dts
+   └── plank.yaml
 
-- A hardware description (usually done by device tree), which specifies embedded
-  SoC reference, connectors and any other hardware components such as LEDs,
-  buttons, sensors or communication peripherals (USB, BLE controller, ...).
+Replace ``plank`` with your board's name, of course.
 
-- A software configuration (done using Kconfig) of features and peripheral
-  drivers. This default board configuration is subordinated to features
-  activation which is application responsibility. Though, it should also enable
-  a minimal set of features common to many applications and to applicable
-  project-provided :ref:`samples-and-demos`.
+The mandatory files are:
 
+#. :file:`plank.dts`: a hardware description in :ref:`devicetree` format. This
+   declares your SoC, connectors, and any other hardware components such as
+   LEDs, buttons, sensors, or communication peripherals (USB, BLE controller,
+   etc).
+
+#. :file:`Kconfig.board`, :file:`Kconfig.defconfig`, :file:`plank_defconfig`:
+   software configuration in :ref:`kconfig` formats. This provides default
+   settings for software features and peripheral drivers.
+
+The optional files are:
+
+- :file:`board.cmake`: used for :ref:`flash-and-debug-support`. You can also
+  use this to build source code files whenever your board is used; see
+  :ref:`build_overview` for details.
+- :file:`index.rst`: documentation for your board. You only need this if you're
+  going to contribute the board to Zephyr.
+- :file:`plank.yaml`: a YAML file with miscellaneous metadata used by the
+  :ref:`sanitycheck_script`.
 
 .. _default_board_configuration:
 
-Default board configuration
-***************************
+Write your devicetree
+*********************
 
-When porting Zephyr to a board, you must provide the board's default
-Kconfig configuration, which is used in application builds unless explicitly
-overridden.
+The devicetree file :file:`boards/<ARCH>/plank/plank.dts` describes your board
+hardware in the Devicetree Source (DTS) format (as usual, change ``plank`` to
+your board's name). If you're new to devicetree, see :ref:`devicetree-intro`.
+
+In general, :file:`plank.dts` should look like this:
+
+.. code-block:: none
+
+   /dts-v1/;
+   #include <your_soc_vendor/your_soc.dtsi>
+
```

Comment:
Would it be reasonable to add a reference to including connector specific .dtsi files here, such as arduino_r3_connector.dtsi? In addition, is it generally not recommended to include other aliases for pins if they don't match a common connector? Several Micropython boards such as the Pyboard and PybNano do not use SoC pin names on their silkscreens.

Reply:
> Would it be reasonable to add a reference to including connector specific .dtsi files here, such as arduino_r3_connector.dtsi?

I can only find those files used in STM32 board definitions, so they appear to just be a convention followed for those boards. Nordic and NXP boards inline the connector node in the main DTS (e.g. nrf52840_pca10056.dts, frdm_k64f.dts).

Since there's not "one way to do it" in DTS, I'm just going to put the `connector` bit into the general recommendations instead:

```
- If your board uses a well-known connector standard (like Arduino, Mikrobus,
  Grove, or 96Boards connectors), add connector nodes to your DTS and configure
  pin muxes accordingly.
```

> In addition, is it generally not recommended to include other aliases for pins if they don't match a common connector?

Can you clarify what you'd like to see changed here? I'm having trouble parsing the double negative ("not recommended [...] if they don't"), sorry.

Reply:
For example, the Pyboard silkscreen uses Y1, Y2 -> Y12 and X1 -> X23 as pin labels. These are not following any existing convention, they're just pin nicknames. In this case, is it best practice to add a `.dtsi` file, or is it on the user to review the documentation of the board and use the actual SoC pin names in their code?

Reply:
I'm not aware of any convention in devicetree where we define macros like Y2 or X23. I spot checked an STM32 board since this thread was originally about one, and didn't see anything there, either (the A0->A5 and D0->D15 silkscreen names are just in comments in the gpio-map property).

Am I missing something?

If you'd like to discuss adding macros or devicetree nodes/properties to make pin numbering in Zephyr more user friendly, @mnkp is the GPIO maintainer and probably the person to talk to. The mailing list or Slack #devicetree channel are a good place to do general discussion unless you want to open a separate issue here on GitHub..

I think feedback from folks like you at Adafruit would be valuable here.

Reply:
I should stress I don't speak for Adafruit, this is all from a personal perspective. 

Let me try and rephrase the pin naming thing: is it possible to support custom aliases for pins in the case that the manufacturer has added nicknames for them that do not match their SoC names? I ask this because the pin naming on a board like the PyBoard is not a "connector" - it's just custom nicknames for pins that are unique to that board revision.

Reply:
> I should stress I don't speak for Adafruit, this is all from a personal perspective.

Sure!

> Let me try and rephrase the pin naming thing: is it possible to support custom aliases for pins in the case that the manufacturer has added nicknames for them that do not match their SoC names? I ask this because the pin naming on a board like the PyBoard is not a "connector" - it's just custom nicknames for pins that are unique to that board revision.

Do you really mean a devicetree alias? Then not to my knowledge, no. In devicetree, a gpio specifier is usually done with a phandle to the GPIO controller plus a specifier (usually a pin and some flags). It's not a node in the devicetree itself. Aliases can only point to nodes.

Reply:
I'll work on some more trial projects and return to this if it seems like it's an obstacle when using hobby style development boards. Thanks for the input.